test/controllers/user.test.ts

```diff
@@ -279,6 +295,49 @@ describe("User controller tests", () => {
 
             expect(user).to.deep.equal(expectedBody);
         });
+
+        it("status 422: returns an appropriate error message if githubUsername isn't a string", async () => {
+            const { body: user } = await chai.request(app).put("/users/" + testUser._id).send({ githubUsername: 1 });
+            const expectedBody = {
+                status: 422,
+                message: "body[githubUsername]: Invalid or missing 'githubUsername'"
+            };
+
+            expect(user).to.deep.equal(expectedBody);
+        });
+
+        it("status 422: returns an appropriate error message if githubToken isn't a string", async () => {
+            const { body: user } = await chai.request(app).put("/users/" + testUser._id).send({ githubToken: 1 });
+            const expectedBody = {
+                status: 422,
+                message: "body[githubToken]: Invalid 'githubToken'"
+            };
+
+            expect(user).to.deep.equal(expectedBody);
+        });
+
+        it("status 200: updates a user email", async () => {
+            const { body: updatedUser } = await chai.request(app).put("/users/" + testUser._id)
+            .send({ email: "example1@gmail.com" })
+            .send({ githubRepo: "https://github.com/CompeteMcgill/challenge-template_new" })
+            .send({ githubToken: "token_new" })
+            .send({ githubUsername: "example_new" })
+            .send({ scores: [] });     
```

Comment:
Sending `scores: []` shouldn't update the scores attribute (see comment on put route)